spec/support/shared_examples/bulk_updatable_type.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+GENERATORS = {
+  string: ->(index) { "wat_#{index}" },
+  boolean: ->(index) { index.even? },
+  hstore: ->(index) { { "foo_#{index}" => "bar_#{index}" } },
+  datetime: ->(index) { Time.zone.now.in_time_zone('UTC').round - index.days },
+  jsonb: ->(index) { { "bee_#{index}" => "bizzle_#{index}" } },
+}.freeze
+
+RSpec.shared_examples 'bulk updatable type' do |type|
+  let(:getter_name) { "#{type}_value" }
+  let(:setter_name) { "#{type}_value=" }
+  let(:type) { type }
+
+  def model_values(models)
+    BulkUpdatableModel.where(id: models.map(&:id)).pluck(getter_name)
+  end
+
+  def generate_values(models, seed: 1)
+    models.each_with_index.map do |model, index|
+      value = GENERATORS.fetch(type).(index * seed)
+      model.public_send(setter_name, value)
+    end
+  end
+
+  it "can update records from null to value for type: #{type}" do
+    values = generate_values(models)
```

Comment:
What's `models` in this context? The one from `spec/junk_drawer/rails/bulk_updatable_spec.rb` line 26?

Reply:
Yep, these examples get executed in the context of the file that they're pulled into.